src/Components/MediaPresenter.ts

```diff
@@ -101,24 +78,16 @@ export class MediaPresenter {
   }
 
   @mobx.action
-  addFilePlayed(fileIndex: number, playedDur: number): void {
-    this.played[fileIndex] =
-      (this.played[fileIndex] + playedDur) % this.durations[fileIndex];
+  resetFilePlayed():void {
```

Comment:
nothing is calling `resetFilePlayed`.... 

I think this might mean your list of `this.played` would be incorrect after first loop. Everything will have been marked as already played.